### ISSUE TITLE
Can not align the metrics and get weried trainning loss curve when trainning with 8*A100

### ISSUE THREAD

**Issue by gaowei724** [HorizonRobotics/Sparse4D#26]
Hi, @linxuewu  thanks you for your excellent work and open source spirit. I encountered a weried problem when training with cfg [sparse4dv3_temporal_r50_1x8_bs6_256x704.py](https://github.com/HorizonRobotics/Sparse4D/blob/main/projects/configs/sparse4dv3_temporal_r50_1x8_bs6_256x704.py): I used A100x8 to train Sparse4Dv3 R50, the training loss declined in the first 18 epochs to about 20, but then it quickly rose. I checked the grad norm and guessed that this might be caused by gradient explosion. I have checked your [training logs](https://github.com/HorizonRobotics/Sparse4D/releases/download/v3.0/sparse4dv3_r50.log) and found no similar situation. In you case , the training loss has been falling to about 11, and then it converged. I noticed that the biggest differences is RTX3090x8 vs A100x8. I'm very confused that could the differences in hardware lead to such differences in training performance? Do you know anything about this problems?

![image](https://github.com/HorizonRobotics/Sparse4D/assets/19744700/ab81f45f-92f2-4887-a6f6-ecc90d5c9e0f)
![image](https://github.com/HorizonRobotics/Sparse4D/assets/19744700/1b0fe771-85de-422a-9b4b-7a3ddee2cb76)
![image](https://github.com/HorizonRobotics/Sparse4D/assets/19744700/ccb0359a-6eea-4f0f-b737-dcd33687da23)

train log (killed on about 20 epochs):
[20240315_173059.log.json](https://github.com/HorizonRobotics/Sparse4D/files/14632138/20240315_173059.log.json)

environments:
Ubuntu 20.04
mmcv (1.7.0) version is different, but I think this is not the key to problem.

Before openning this issue, I also read [training divergence](https://github.com/HorizonRobotics/Sparse4D/issues/4)

**Comment by linxuewu** [comment]
I can train using a800*8. In theory, there shouldn't be much difference with a100. 

Please check the following: 1. Whether the backbone pretrain checkpoint has been correctly loaded, 2. Whether the latest code has been pulled, 3. Whether the preparation work according to ['quick_start.md'](https://github.com/HorizonRobotics/Sparse4D/blob/main/docs/quick_start.md) has been completed.

**Comment by linxuewu** [comment]
Another point to note is that the environment for compiling deformable aggregation needs to be consistent with the training environment

**Comment by gaowei724** [comment]
> I can train using a800*8. In theory, there shouldn't be much difference with a100.
> 
> Please check the following: 1. Whether the backbone pretrain checkpoint has been correctly loaded, 2. Whether the latest code has been pulled, 3. Whether the preparation work according to ['quick_start.md'](https://github.com/HorizonRobotics/Sparse4D/blob/main/docs/quick_start.md) has been completed.

Thank you for your suggestion. It turned out that the issue was due to the use of outdated v3 code. After pulling the latest main branch, I can now train [sparse4dv3_temporal_r50_1x8_bs6_256x704.py](https://github.com/HorizonRobotics/Sparse4D/blob/main/projects/configs/sparse4dv3_temporal_r50_1x8_bs6_256x704.py) normally. I'm currently conducting experiments comparing with and without denoise, but even when I turn off denoise in the [sparse4dv3_temporal_r50_1x8_bs6_256x704.py](https://github.com/HorizonRobotics/Sparse4D/blob/main/projects/configs/sparse4dv3_temporal_r50_1x8_bs6_256x704.py) with latest main branch (my approach is to set num_dn_groups=0, with all other settings unchanged), I still encounter gradient explosion that leads to training divergence. Could you please provide any clues for this problem? Have you conducted experiments with denoise turned off before and what's your setting?

**Comment by linxuewu** [comment]
If 'dn' is removed, please increase the total batch size to 64, or appropriately reduce the learning rate.

**Comment by gaowei724** [comment]
> If 'dn' is removed, please increase the total batch size to 64, or appropriately reduce the learning rate.

Thx. This solved the problem.

**Comment by owphoo** [comment]
> > If 'dn' is removed, please increase the total batch size to 64, or appropriately reduce the learning rate.
> 
> Thx. This solved the problem.

@gaowei724 hello, in you experiment, after removing **dn**, were **tracking ids** still consistent?
